transit_predictions_511.py

```diff
@@ -280,36 +64,19 @@ def get_predictions(self):
             - <prediction 1> <prediction 2> ...
         """
 
-        print(f'Getting predictions for agency {self.__agency} for stop_code {self.__stop_code}')
-
-        self.__poll()
-
-        # Polling failed
-        if not self.__data:
-            return ['Network Error', f'{self.__status_code}', f'{self.__reason}']
-
-        predictions = JsonHandler.predictions_for_route_codes(self.__data, self.__route_codes, self.__direction)
+        predictions = self.__data_handler.predictions_for_route_codes(
+            self.__data,
+            self.__route_codes,
+            self.__direction
+        )
         prediction_text = []
 
         if predictions:
             for route_code in self.__route_codes:
                 if route_code in predictions:
                     route = predictions[route_code]
 
-                    prediction_text.append(f'{route_code} {route.title}')
-
-                    prediction_count = len(route.predictions)
-                    n = min(self.__max_predictions, prediction_count)
-
-                    for i in range(n):
-                        route.predictions[i] = f'{route.predictions[i]}m'
-
-                    if route.predictions[0] == '0m':
-                        route.predictions[0] = 'Now'
-
-                    route_predictions = ' '.join(route.predictions[:n])
-
-                    prediction_text.append(route_predictions)
+                    prediction_text.extend(self.__display.formatter.format(route))
 
```

Comment:
Add logs if no predictions

Reply:
That's in line 87 of the newer version of the file